### PR TITLE
feat: add support for The Isle Evrima

### LINKED ISSUE #500: feat: Add support for The Isle Evrima
**What is this feature about?**
Add support for The Isle Evrima game using the EOS protocol

**Additional context/references**
We were able to obtain the credentials to perform the query on The Isle Evrima servers, which is a variation of the legacy game.

I did some tests on random public servers like this:

IP: 50.18.155.230
Port: 7777

```sh
node .\examples\simple.js > eos.json
```

```json
{
    "name": "",
    "raw": {
        "deployment": "6db6bea492f94b1bbdfcdfe3e4f898dc",
        "id": "c73409012a5b410fa4c7fc77ff96e7d8",
        "bucket": "<None>:<None>:<None>",
        "settings": {
            "maxPublicPlayers": 115,
            "allowInvites": true,
            "shouldAdvertise": true,
            "allowReadById": true,
            "allowJoinViaPresence": true,
            "allowJoinInProgress": true,
            "allowConferenceRoom": false,
            "checkSanctions": false,
            "allowMigration": false,
            "rejoinAfterKick": "",
            "platforms": null
        },
        "totalPlayers": 41,
        "openPublicPlayers": 74,
        "publicPlayers": [],
        "started": true,
        "lastUpdated": "2024-01-21T18:54:30.054Z",
        "attributes": {
            "SERVERNAME_s": "Official Evrima - NA 6",
            "PASSWORD_s": "E04EE3DC8923189B92455EC2807284AF1FD372F5",
            "QUEUE_ENABLED_b": true,
            "PAGINATION_s": "0",
            "SERVER_VERSION_s": "0.13.21.23",
            "PASSWORD_ENABLED_b": false,
            "QUEUE_PORT_s": "10000",
            "ADDRESS_s": "50.18.155.230",
            "__EOS_BLISTENING_b": true,
            "MAP_NAME_s": "Gateway",
            "ADDRESSBOUND_s": "0.0.0.0:7777",
            "IP_s": "50.18.155.230",
            "__EOS_BUSESPRESENCE_b": true,
            "DISCORD_s": "https://discord.gg/KPtBfwD7qw",
            "OFFICIAL_b": true
        },
        "owner": "Client_xyza78910qHhuFMTLERSgsk7E97WeHom",
        "ownerPlatformId": null
    },
    "maxplayers": 115,
    "numplayers": 41,
    "players": [],
    "bots": [],
    "queryPort": 7777,
    "connect": "50.18.155.230:7777",
    "ping": 0
}
```

However, there are still some problems, such as the server name, which does not use the same attribute as ASA `CUSTOMSERVERNAME_s` or Palworld `NAME_s` #497.

Given the variation from game to game, perhaps it would be better to pass this part of obtaining the correct attributes to the protocol for each game, where the base protocol returns the raw response to the game then populates the stable values.

### PR DESCRIPTION
**Pull request by GuilhermeWerner**
### This PR adds

Support for The Isle Evrima using EOS protocol.

Closes #500 

I also moved the server and map name assignment to the specific protocol of The Isle.

It will still be necessary to modify the ASA and Palworld protocols to obtain this correct attribute data, I omitted this from this PR so as not to mix the games in the same modification.

### Tests:

Keep in mind that there are servers with the legacy version of The Isle, which do not work with this protocol, check if you are testing with a server of the correct version.

A random official server:

```js
GameDig.query({
  type: 'theisleevrima',
  host: '50.18.155.230',
  port: '7777'
}).then((state) => {
  console.log(JSON.stringify(state))
}).catch((error) => {
  console.log(`Server is offline, error: ${error}`)
})
```

```json
{
    "name": "Official Evrima - NA 6",
    "map": "Gateway",
    "password": false,
    "raw": {
        "deployment": "6db6bea492f94b1bbdfcdfe3e4f898dc",
        "id": "c73409012a5b410fa4c7fc77ff96e7d8",
        "bucket": "<None>:<None>:<None>",
        "settings": {
            "maxPublicPlayers": 115,
            "allowInvites": true,
            "shouldAdvertise": true,
            "allowReadById": true,
            "allowJoinViaPresence": true,
            "allowJoinInProgress": true,
            "allowConferenceRoom": false,
            "checkSanctions": false,
            "allowMigration": false,
            "rejoinAfterKick": "",
            "platforms": null
        },
        "totalPlayers": 98,
        "openPublicPlayers": 17,
        "publicPlayers": [],
        "started": true,
        "lastUpdated": "2024-01-21T20:15:39.682Z",
        "attributes": {
            "SERVERNAME_s": "Official Evrima - NA 6",
            "PASSWORD_s": "E04EE3DC8923189B92455EC2807284AF1FD372F5",
            "QUEUE_ENABLED_b": true,
            "PAGINATION_s": "0",
            "SERVER_VERSION_s": "0.13.21.23",
            "PASSWORD_ENABLED_b": false,
            "QUEUE_PORT_s": "10000",
            "ADDRESS_s": "50.18.155.230",
            "__EOS_BLISTENING_b": true,
            "MAP_NAME_s": "Gateway",
            "ADDRESSBOUND_s": "0.0.0.0:7777",
            "IP_s": "50.18.155.230",
            "__EOS_BUSESPRESENCE_b": true,
            "DISCORD_s": "https://discord.gg/KPtBfwD7qw",
            "OFFICIAL_b": true
        },
        "owner": "Client_xyza78910qHhuFMTLERSgsk7E97WeHom",
        "ownerPlatformId": null
    },
    "maxplayers": 115,
    "numplayers": 98,
    "players": [],
    "bots": [],
    "queryPort": 7777,
    "connect": "50.18.155.230:7777",
    "ping": 0
}
```

### CODE REVIEW

File: lib/games.js [hunk 1]
```diff
@@ -1328,6 +1328,14 @@ export const games = {
     },
     release_year: 2015
   },
+  theisleevrima: {
```
Comment (CosminPerRam) [inline lib/games.js:1331]: CI returns
```
IDFail {
    game_id: "theisleevrima",
    game_name: "The Isle Evrima",
    expected_id: "tie",
    rule_stack: [
        MoreThanTwoWordsMakeAcronym,
    ],
}
```
so the gid should be `tie`.

**Review by CosminPerRam — CHANGES_REQUESTED** [review]
LGTM, thanks a lot!
Please also add a line to the changelog file (:

File: lib/games.js (hunk 1 above)
Comment (GuilhermeWerner) [reply lib/games.js:1331]: Should I change just the gid or the protocol too? The file name from `theisleevrima.js` to `tie.js`

Comment (CosminPerRam) [reply lib/games.js:1331]: Just the gid, the protocol could be explicit about its name (:

File: protocols/epic.js [hunk 2]
```diff
@@ -126,9 +126,6 @@ export default class Epic extends Core {
       throw new Error('Server not found')
     }
 
-    state.name = desiredServer.attributes.CUSTOMSERVERNAME_s
-    state.map = desiredServer.attributes.MAPNAME_s
-    state.password = desiredServer.attributes.SERVERPASSWORD_b
```
Comment (CosminPerRam) [inline protocols/epic.js:131]: These could remain here and only overwrite the values if needed (MAPNAME_s is the same for asa and this one).

Comment (GuilhermeWerner) [reply protocols/epic.js]: In this case it is `MAP_NAME_s`, in the ASA it is `MAPNAME_s`. I thought the epic would be a little more stable in that sense, but it seems that most of the attributes are game dependent.

Comment (CosminPerRam) [reply protocols/epic.js]: Oh, oops, well they could be default values then.

**Review by CosminPerRam — APPROVED** [review]

**Pull request merged**